app/services/backup.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+class Backup
+  def initialize(user)
+    @user = user
+  end
+
+  def run
+    generate_json_files
+    generate_zip_file
+  end
+
+  def backup_zip_dir
+    Rails.root.join('tmp', 'backup', 'zip')
+  end
+
+  def path_to_zip
+    Rails.root.join(backup_zip_dir, "#{@user.name}_backup.zip")
```

Comment:
`backup_zip_dir` の返り値は `Pathname` オブジェクトなので、 そのままjoin可能

```suggestion
    backup_zip_dir.join("#{@user.name}_backup.zip")
```